pkgs/development/tools/documentation/doxygen/default.nix

```diff
@@ -15,30 +15,34 @@
 
 stdenv.mkDerivation rec {
   pname = "doxygen";
-  version = "1.10.0";
+  version = "1.12.0";
 
   src = fetchFromGitHub {
     owner = "doxygen";
     repo = "doxygen";
     rev = "Release_${lib.replaceStrings [ "." ] [ "_" ] version}";
-    sha256 = "sha256-FPI5ICdn9Tne/g9SP6jAQS813AAyoDNooDR/Hyvq6R4=";
+    hash = "sha256-4zSaM49TjOaZvrUChM4dNJLondCsQPSArOXZnTHS4yI=";
   };
 
-  patches = [
-    (fetchpatch {
-      name = "sys-spdlog-fix.patch";
-      url = "https://github.com/doxygen/doxygen/commit/0df6da616f01057d28b11c8bee28443c102dd424.patch";
-      hash = "sha256-7efkCQFYGslwqhIuPsLYTEiA1rq+mO0DuyQBMt0O+m0=";
-    })
-  ];
-
   nativeBuildInputs = [
     cmake
     python3
     flex
     bison
   ];
 
+  # https://github.com/doxygen/doxygen/issues/10928#issuecomment-2179320509
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace-fail 'JAVACC_CHAR_TYPE=\"unsigned char\"' \
+                     'JAVACC_CHAR_TYPE=\"char8_t\"' \
+      --replace-fail "CMAKE_CXX_STANDARD 17" "CMAKE_CXX_STANDARD 20"
+  '';
+
+  env = lib.optionalAttrs stdenv.cc.isClang {
+    CXXFLAGS = "-Wno-error=missing-template-arg-list-after-template-kw";
+  };
```

Comment:
Let’s apply the upstream fix https://github.com/doxygen/doxygen/commit/cff64a87dea7596fd506a85521d4df4616dc845f instead, so that we’re forced to remove it on the next release rather than have this hang around forever.